### PR TITLE
chore: update issue and pr templates

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #359

This pull request adds pull request templates and updates issue templates based on those in cosmos-sdk.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] confirmed that this PR does not change production code

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+### Author Checklist
+
+*All items are required. Please add a note to the item if the item is not applicable and
+please add links to any relevant follow up issues.*
+
+I have...
+
+- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
+- [ ] added `!` to the type prefix if API or client breaking change
+- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
```
Comment (ryanchristo) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: We don't have documentation for targeting the correct branch within this repository but I think its ok to link to the cosmos-sdk contributing documentation unless we want to define a different process for regen ledger.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+- [ ] provided a link to the relevant issue or specification
+- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
```
Comment (ryanchristo) [inline .github/PULL_REQUEST_TEMPLATE.md:28]: Same as above. I think it still makes sense to link to cosmos-sdk documentation here.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
```
Comment (ryanchristo) [inline .github/PULL_REQUEST_TEMPLATE.md:29]: Same as above. I think it still makes sense to link to cosmos-sdk documentation here.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (robert-zaremba) [reply .github/PULL_REQUEST_TEMPLATE.md:26]: yes, no need to duplicate. If it will be updated - we will only update in one place.

**Review by robert-zaremba — APPROVED** [review]

**Review by likhita-809 — APPROVED** [review]
looks good!

**Pull request merged**